packages/api-explorer/src/DocAsync.jsx

```diff
@@ -0,0 +1,55 @@
+const React = require('react');
+const { useEffect, useState } = require('react');
+const PropTypes = require('prop-types');
+
+const Doc = require('./Doc');
+const ErrorBoundary = require('./ErrorBoundary');
+
+const DocAsync = props => {
+  const { maskErrorMessages, oas, onError, ...docProps } = props;
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState();
+
+  useEffect(() => {
+    (async () => {
+      try {
+        await oas.dereference();
+
+        setLoading(false);
+      } catch (err) {
+        setError(err);
+      }
+    })();
+  }, [oas]);
+
+  if (error) {
+    // We're passing in `error.message` into the error boundary because it expects a child node to be present. It's fine
+    // though as since we're also passing an `error` prop we'll default to our standard error handling instead of
+    // rendering out that error message.
+    return (
+      <ErrorBoundary appContext="explorer" error={error} maskErrorMessages={maskErrorMessages} onError={onError}>
+        {error.message}
+      </ErrorBoundary>
+    );
+  }
+
+  if (loading) {
+    return null;
```

Comment:
Maybe this is where we can render a stubbed version of the <Doc> to improve perceived performance in future?

Reply:
Yeah a loading spinner, or maybe something like Facebook does where they load in opaque boxes in that get swapped out as components load.